src/components/element-editor/conditional-rendering/conditional-rendering.tsx

```diff
@@ -26,21 +23,22 @@ export const ConditionalRendering = () => {
     // =========================================================================
 
     const { focusedElement, elements, updateFocusedElement } = useBuilder();
-    const element = focusedElement.element;
-    const [childEntryValues, setChildEntryValues] =
-        useState<IConditionalRendering[]>();
+    const element = focusedElement?.element;
+    const [, setChildEntryValues] = useState<IConditionalRendering[]>([]);
     const {
         setValue,
         formState: { isDirty },
+        watch,
         getValues,
     } = useFormContext<IBaseTextBasedFieldValues>();
-    const shouldUpdateFocusedElement =
-        isDirty ||
-        childEntryValues?.length >
-            focusedElement?.element?.conditionalRendering?.length;
 
     const schema = SchemaHelper.buildSchema(EElementType.EMAIL);
     const invalidAndEmptyFields = getTouchedAndErrorsFields();
+    const conditionalRenderingValues = getValues("conditionalRendering") || [];
+    const shouldUpdateFocusedElement =
+        isDirty ||
+        conditionalRenderingValues?.length >
+            focusedElement?.element?.conditionalRendering?.length;
```

Comment:
can we update the focused element's dirty state based on `isDirty` flag alone?

I notice the form is not dirty on adding / removing entry, you can use `{shouldDirty: true}` in `setValue()` since the form has indeed been updated

if we can, this useEffect is not necessary since `BasicDetails` is already updating based on the dirty state